### ISSUE TITLE
(WCAG 2.1) Announce when errors are present to screen reader users

### ISSUE THREAD

**Issue by jeanniefu** [civiform/civiform#3563]
Required to meet WCAG 2.1 guidelines. Identified during an accessibility audit against WCAG 2.1.

**WCAG SC**:  [4.1.3. Status messages](https://www.w3.org/TR/WCAG21/#status-messages)

**Observed results**: There is no announcement by screen reader that errors are present.

**How to fix**: 
From a11y team:
Assign appropriate ARIA roles or aria-live properties to status messages so that the new content is spoken to users of screen readers. The element which acts as the live region needs to be present in the document object model (DOM) prior to the content being inserted to make sure the change is detected by some screen readers. Status message can be visible or not. If visible, these should be at the top of the form.

Examples:
```
<div role=""status"">Fix errors in the form before you can proceed</div> OR
<div class=""sr-only"" aria-live=""polite"">Address fields in error before proceeding</div>
```

From me:
This needs additional scoping for the proper solution. I think file upload errors can use `aria-live` (since errors are handled client side). All other error messages actually are rendered server-side, so we'll need to look more into whether using `aria-live` works here. From a quick skim of the w3.org docs, one approach that works is "to include an error notification in the page title (title element)" ([source](https://www.w3.org/WAI/WCAG21/Techniques/general/G83#description)).

**Comment by jeanniefu** [comment]
Related issue: #3287. Should evaluate how this feedback fits together and what the correct solution is.

**Comment by jeanniefu** [comment]
My proposal:

- File upload is client side validation. I will use `aria-live` to announce that status change to the screen reader.
- Others are all server side validation, and reload the page with errors. I will set sr-only text, "There are errors in the form. Please fix before continuing.", with `role-status` to announce that to the screen reader on page reload. This text will not be visible. 

Will check with UX if that text is OK.

**Comment by jeanniefu** [comment]
@mferhut - Does using this text sound good? "There are errors in the form. Please fix before continuing."

**Comment by mferhut** [comment]
Do we provide various messages per number of errors? 

1. If there are more than text field error we can be general and say "There are errors in the form. Please fix before continuing." Also, when the user gets to the error field what does the screen reader say? ex. "Please enter a valid email."
2. If there is only one text field error, can we take the user to that specific field to fix and provide a message? "Please enter a valid email before continuing." If we cannot, then let's stick with #1 scenario. 

Let me know if you have any questions. Thanks!

**Comment by jeanniefu** [comment]
Thanks that sounds good!

We have visual text for field errors. When the user gets to the field with an error, the screen reader will read that text (e.g. the red text in this screenshot):

<img width="1237" alt="Screen Shot 2022-10-21 at 12 00 57 PM" src="https://user-images.githubusercontent.com/9094240/197270432-c0ebaed2-5540-4031-8564-711633a4c9f3.png">

**Comment by mferhut** [comment]
Thanks for the screenshot.

**Comment by jeanniefu** [comment]
During AR testing, we had feedback that the error message wasn't prominent enough. One option is to change the setting of the `aria-live` to `assertive`, so the message interrupts the current screen reader text.
